src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -65,39 +67,60 @@ public EcheladderScreen()
 		guiWidth = 250;
 		guiHeight = 202;
 		
-		maxScroll = (ClientRungData.getFinalRungIndex() + 1) * RUNG_Y - 154;
+		maxScroll = ClientRungData.getFinalRungIndex() - 10;
 	}
 	
 	@Override
 	public void init()
 	{
 		super.init();
-		scrollIndex = Mth.clamp((ClientPlayerData.getRung() - 8) * RUNG_Y, 0, maxScroll);
 		animatedRung = Math.max(animatedRung, lastRung);    //If you gain a rung while the gui is open, the animated rung might get higher than the lastRung. Otherwise they're always the same value.
 		fromRung = lastRung;
 		lastRung = ClientPlayerData.getRung();
+		
+		rungBars.clear();
+		for(int i = 0; i < ClientRungData.getFinalRungIndex() + 1; i++)
+		{
+			Component name = Component.translatable(I18n.exists("echeladder.rung." + i) ? I18n.get("echeladder.rung." + i) : "Rung " + (i + 1));
+			
+			Optional<String> tooltip = Optional.empty();
+			String description = ClientRungData.getData(i).description().orElse("");
+			if(!description.isEmpty())
+				tooltip = Optional.of(description);
```

Comment:
Seems excessive. Is there a particular reason that you structured it this way?
```suggestion
			Optional<String> tooltip = ClientRungData.getData(i).description();
```